src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -970,11 +975,32 @@ public String getName() {
         @Override
         public GroupDetailsImpl mapAttributes(String dn, Attributes attributes) throws NamingException {
             LdapName name = new LdapName(dn);
-            String groupName = fixGroupname(String.valueOf(name.getRdn(name.size() - 1).getValue()));
+            String groupName = fixGroupname(extractGroupName(name, attributes));
             return new GroupDetailsImpl(dn, groupName);
         }
-    }
 
+        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            String groupName = StringUtils.EMPTY;
+            if (!GET_CN_FROM_ATTRIBUTES) {
+                groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            } else {
+                for (NamingEnumeration ae = attributes.getAll(); ae.hasMore(); ) {
```

Comment:
I think you can just use `Attribute attr = attributes.get(CN)` with a null check on the result and get rid of the loop.

Reply:
See https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fldap-plugin/detail/PR-36/6/pipeline

> [2019-09-26T07:22:30.811Z] [INFO] Nullcheck of ae at line 987 of value previously dereferenced in hudson.security.LDAPSecurityRealm$GroupDetailsMapper.extractGroupName(LdapName, Attributes) [hudson.security.LDAPSecurityRealm$GroupDetailsMapper, hudson.security.LDAPSecurityRealm$GroupDetailsMapper] At LDAPSecurityRealm.java:[line 987]Redundant null check at LDAPSecurityRealm.java:[line 987] RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE